### PR TITLE
Externalize JavaScript: download export

### PR DESCRIPTION
**Pull request by orangejenny**
Just one page in this PR; it took a while to figure out how to get `djng_current_rmi` added to the view context. Will probably move the `get_context_data` override into a mixin that can be used by other angular views, but decided it was premature to do that now.

@esoergel / @biyeun

### CODE REVIEW

File: corehq/apps/export/static/export/js/download_export.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+hqDefine('export/js/download_export.js', function() {
+   'use strict';
```
Comment (dimagimon) [inline corehq/apps/export/static/export/js/download_export.js]: Expected indentation of 4 space characters but found 3. (indent)

File: corehq/apps/export/static/export/js/download_export.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    downloadExportsApp.constant('formElement', {
+        progress: function () {
+            return $('#download-progress-bar');
+        },
+        group: function () {
+            return $('#id_group');
+        },
+        user_type: function () {
+            return $('#id_user_types');
+        }
```
Comment (dimagimon) [inline corehq/apps/export/static/export/js/download_export.js]: Missing trailing comma. (comma-dangle)

File: corehq/apps/export/views.py [hunk 3]
```diff
@@ -570,6 +570,15 @@ def page_context(self):
...
         return context
 
+    # Add the output of djng_current_rmi to view context, which requires having
+    # the rest of the context, specifically context['view'], available.
+    # See https://github.com/jrief/django-angular/blob/master/djng/templatetags/djng_tags.py
+    def get_context_data(self, **kwargs):
+        context = super(BaseDownloadExportView, self).get_context_data(**kwargs)
+        from djangular.templatetags.djangular_tags import djng_current_rmi
+        context['djng_current_rmi'] = json.loads(djng_current_rmi(context))
+        return context
```
Comment (esoergel) [inline corehq/apps/export/views.py]: Is this something that should always be available?  Will we need it any time we want to have fully external JS and RMI?  If so, I'm wondering if we should make our own subclass of `JSONResponseMixin` (with a better name), and include this there.

Comment (esoergel) [inline corehq/apps/export/views.py]: Oh, I just saw your comment in the PR description along those lines.  Sounds fair.

**Review by esoergel — APPROVED** [review]

**Pull request merged**